packages/edit-site/src/components/global-styles/navigation-button.js

```diff
@@ -7,18 +7,29 @@ import {
 	FlexItem,
 	__experimentalHStack as HStack,
 } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
 import { Icon } from '@wordpress/icons';
 
 function NavigationButton( {
 	path,
 	icon,
 	children,
+	label: labelProp,
 	isBack = false,
 	...props
 } ) {
 	const navigator = useNavigator();
+
+	const defaultLabel = isBack
+		? __( 'Navigate to the previous screen' )
+		: undefined;
+
 	return (
-		<Item onClick={ () => navigator.push( path, { isBack } ) } { ...props }>
+		<Item
+			onClick={ () => navigator.push( path, { isBack } ) }
+			aria-label={ labelProp ?? defaultLabel }
```

Comment:
Should this be `label` prop like the `Button` component. I also wonder why it's not taking the "content" as a label like it's supposed to do by default for buttons.

Reply:
> Should this be label prop like the Button component.

The `Item` component renders a vanilla HTML `button` instead of a WordPress `Button` (sorry if I caused any confusion in a [previous comment](https://github.com/WordPress/gutenberg/pull/35142#discussion_r716728303) where I stated the opposite).

> I also wonder why it's not taking the "content" as a label like it's supposed to do by default for buttons.

I believe it is, but in the case of the back buttons in the Global Styles sidebar, these buttons [only have an icon as their content](https://github.com/WordPress/gutenberg/blob/91ea5649aa95bab04c6fb3977b0a6ac1bff189c7/packages/edit-site/src/components/global-styles/header.js#L24-L34) (no human-readable text)

Reply:
Got it Thanks, should we use a `label` prop to align with `Button` or keep the raw `aria-label`? I think it's a bit mixed right now in the different components we have and not sure what's the best path forward (maybe a small preference for `label` in components where it's a generic use-case for compatibility with mobile).

Reply:
I agree that it would be nice to have a less mixed approach to how we pass and name props related to labels.

Given the fact that `Item` renders a HTML `button`, at the moment we have to pass the `aria-label` prop to `Item` (since `label` is not a standard HTML attribute).
Maybe a way to improve the situation would be to rename the `label` prop on the `NavigationButton` component to `aria-label`? I could also remove the changes from the `NavigationButton` and pass the `aria-label` attribute directly from the `ScreenHeader` component:

```diff
diff --git a/packages/edit-site/src/components/global-styles/header.js b/packages/edit-site/src/components/global-styles/header.js
index 1ddd562aa2..1496bb440c 100644
--- a/packages/edit-site/src/components/global-styles/header.js
+++ b/packages/edit-site/src/components/global-styles/header.js
@@ -31,6 +31,7 @@ function ScreenHeader( { back, title, description } ) {
 						}
 						size="small"
 						isBack
+						aria-label={ __( 'Navigate to the previous screen' ) }
 					/>
 				</View>
 				<Spacer>
```


Separately from this PR, we could consider rendering a WordPress `Button` instead of a HTML `button` inside `Item`, which could help to make our APIs a bit more coherent

Reply:
> I could also remove the changes from the NavigationButton and pass the aria-label attribute directly from the ScreenHeader component

I went ahead and applied that suggested change in [`a528e22` (#35325)](https://github.com/WordPress/gutenberg/pull/35325/commits/a528e2242acc93cbab0834a09e3ab8b7c5b69b2e)